src/mango/src/mango_idx.erl

```diff
@@ -70,13 +70,33 @@ get_usable_indexes(Db, Selector0, Opts) ->
         ?MANGO_ERROR({no_usable_index, no_index_matching_name})
     end,
 
-    SortIndexes = mango_idx:for_sort(FilteredIndexes, Opts),
-    if SortIndexes /= [] -> ok; true ->
-        ?MANGO_ERROR({no_usable_index, missing_sort_index})
+    UsableFilter = fun(I) -> mango_idx:is_usable(I, Selector) end,
+    UsableIndexes0 = lists:filter(UsableFilter, FilteredIndexes),
+
+    UsableIndexes1 = case has_use_index(Opts) of
+        true ->
+            UsableFilteredIndexes = mango_cursor:maybe_filter_indexes_by_ddoc(UsableIndexes0, Opts),
```

Comment:
maybe this can be written as:
```
case mango_cursor:maybe_filter_indexes_by_ddoc(UsableIndexes0, Opts) of
    [] -> ?MANGO_ERROR({no_usable_index, no_usable_index_matching_name});
    UsableFilteredIndexes -> UsableFilteredIndexes
end;
```